cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+
+    ```bash
+    rad resource-provider list
+    ```
+    The list of resource providers are displayed.
+
+    ```bash
+    rad resource-type list --resource-provider Mycompany.Messaging
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `plaidResource` and adds to the resource provider
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    ```
+
+6. Deb wants to know the details of resource provider
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    ```
+    The details of the resource provider `Mycompany.Messaging` are displayed.
+
+7. Deb deletes a resource-type Plaid from the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-type delete Mycompany.Messaging/plaidResource
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource type plaidResource from the resource provider Mycompany.Messaging? (yes/no)
+    ```
+    The resource type `Plaid` is deleted from the resource provider `Mycompany.Messaging`
+
+8. Deb deletes the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource provider Mycompany.Messaging? (yes/no)
+    ```
+
+## Finalized set of commands
+
+### rad resource-provider
+
+#### 1. rad resource-provider create
+
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Creating a resource provider involves creating new resource types that can be used in your applications. Inputs can be passed in as a file or as flags.
+
+Create a resource provider from a file 
+    
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    Resource provider Mycompany.Messaging created successfully
+    ```
+**Flags**
+- `-f`, `--fom-file` : The file that contains the resource provider definition
+
+#### 2. rad resource-provider list
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can list all the resource providers available in your tenant.
+
+    ```bash
+    rad resource-provider list 
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 3. rad resource-provider show
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. User can display the details of the resource providers.
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    RESOURCE-PROVIDER   TYPE                                STATE
```

Comment:
Today, the column heading for the `name` is always `RESOURCE` regardless of the resource type. We could change this in the future, but `RESOURCE` would be consistent with our other commands.

Reply:
check this across other commands - app, env, workspace, resources